server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2644,6 +2643,9 @@ private void verifyVmLimits(UserVmVO vmInstance, Map<String, String> details) {
             Map<String, String> customParameters = new HashMap<>();
             customParameters.put(VmDetailConstants.CPU_NUMBER, String.valueOf(newCpu));
             customParameters.put(VmDetailConstants.MEMORY, String.valueOf(newMemory));
+            if (svcOffering.getSpeed() == null) {
```

Comment:
Maybe this check isn't required, as it is already present in validateCustomParameter(), however no biggie